File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/screens/payments/PaymentsScreen.kt
package com.gym.gymapp.ui.screens.payments

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowUpward
import androidx.compose.material.icons.filled.History
import androidx.compose.material.icons.filled.Payments
import androidx.compose.material.icons.filled.PendingActions
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun PaymentsScreen() {
    val scrollState = rememberScrollState()

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .verticalScroll(scrollState)
            .padding(16.dp)
    ) {
        Spacer(Modifier.height(20.dp))
        Text(
            "Revenue & Payments",
            style = MaterialTheme.typography.headlineSmall,
            fontWeight = FontWeight.Black,
            modifier = Modifier.padding(bottom = 24.dp)
        )

        // Summary Cards
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            SummaryCard(
                title = "Total Revenue",
                amount = "₹1,24,500",
                icon = Icons.Default.Payments,
                color = Color(0xFF10B981),
                modifier = Modifier.weight(1f)
            )
            SummaryCard(
                title = "Pending Dues",
                amount = "₹12,400",
                icon = Icons.Default.PendingActions,
                color = Color(0xFFF59E0B),
                modifier = Modifier.weight(1f)
            )
        }

        Spacer(Modifier.height(24.dp))

        // Recent Transactions Section
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text("Recent Transactions", fontWeight = FontWeight.Bold, fontSize = 16.sp)
            TextButton(onClick = {}) {
                Text("View All", fontSize = 12.sp)
            }
        }

        Spacer(Modifier.height(8.dp))

        // Transaction List
        TransactionItem("Muhammed Ansib", "Monthly Plan", "₹2,500", "Mar 18, 02:30 PM", true)
        TransactionItem("Rahul Sharma", "Personal Training", "₹5,000", "Mar 17, 11:15 AM", true)
        TransactionItem("John Doe", "Quarterly Plus", "₹6,500", "Mar 17, 09:45 AM", true)
        TransactionItem("Adarsh S", "Basic Plan", "₹1,200", "Mar 16, 04:20 PM", false)
        TransactionItem("Vignesh K", "Yearly Pack", "₹12,000", "Mar 15, 10:00 AM", true)
    }
}

@Composable
fun SummaryCard(title: String, amount: String, icon: ImageVector, color: Color, modifier: Modifier = Modifier) {
    Card(
        modifier = modifier,
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)
    ) {
        Column(Modifier.padding(16.dp)) {
            Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(24.dp))
            Spacer(Modifier.height(12.dp))
            Text(title, fontSize = 11.sp, color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f))
            Text(amount, fontSize = 18.sp, fontWeight = FontWeight.Black, color = MaterialTheme.colorScheme.onSurface)
        }
    }
}

@Composable
fun TransactionItem(name: String, plan: String, amount: String, date: String, isSuccess: Boolean) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Surface(
                    modifier = Modifier.size(40.dp),
                    shape = CircleShape,
                    color = (if (isSuccess) Color(0xFF10B981) else Color.Red).copy(alpha = 0.1f)
                ) {
                    Box(contentAlignment = Alignment.Center) {
                        Icon(
                            if (isSuccess) Icons.Default.ArrowUpward else Icons.Default.History,
                            contentDescription = null,
                            tint = if (isSuccess) Color(0xFF10B981) else Color.Red,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
                Spacer(Modifier.width(12.dp))
                Column {
                    Text(name, fontWeight = FontWeight.Bold, fontSize = 14.sp)
                    Text(plan, fontSize = 11.sp, color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f))
                }
            }
            Column(horizontalAlignment = Alignment.End) {
                Text(amount, fontWeight = FontWeight.Black, fontSize = 14.sp, color = if (isSuccess) Color(0xFF10B981) else Color.Red)
                Text(date, fontSize = 10.sp, color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f))
            }
        }
    }
}

File: gym_flutter_app/lib/ui/screens/payments_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../logic/providers/dashboard_provider.dart';

class PaymentsScreen extends ConsumerWidget {
  const PaymentsScreen({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final uiState = ref.watch(dashboardProvider);
    final theme = Theme.of(context);

    return Scaffold(
      backgroundColor: theme.colorScheme.surface,
      body: SingleChildScrollView(
        padding: const EdgeInsets.symmetric(horizontal: 16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 60),
            Text(
              "Revenue & Payments",
              style: theme.textTheme.headlineSmall?.copyWith(
                fontWeight: FontWeight.w900,
                color: theme.colorScheme.onSurface,
              ),
            ),
            const SizedBox(height: 24),
            Row(
              children: [
                Expanded(
                  child: _SummaryCard(
                    title: "Total Revenue",
                    amount: uiState.totalRevenue,
                    icon: Icons.payments,
                    color: const Color(0xFF10B981),
                  ),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: _SummaryCard(
                    title: "Pending Dues",
                    amount: "₹12,400",
                    icon: Icons.pending_actions,
                    color: const Color(0xFFF59E0B),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 24),
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Text("Recent Transactions", style: theme.textTheme.titleSmall?.copyWith(fontWeight: FontWeight.bold)),
                TextButton(onPressed: () {}, child: Text("View All", style: TextStyle(fontSize: 12, color: theme.colorScheme.primary))),
              ],
            ),
            const SizedBox(height: 8),
            const _TransactionItem(name: "Muhammed Ansib", plan: "Monthly Plan", amount: "₹2,500", date: "Mar 18, 02:30 PM", isSuccess: true),
            const _TransactionItem(name: "Rahul Sharma", plan: "Personal Training", amount: "₹5,000", date: "Mar 17, 11:15 AM", isSuccess: true),
            const _TransactionItem(name: "John Doe", plan: "Quarterly Plus", amount: "₹6,500", date: "Mar 17, 09:45 AM", isSuccess: true),
            const _TransactionItem(name: "Adarsh S", plan: "Basic Plan", amount: "₹1,200", date: "Mar 16, 04:20 PM", isSuccess: false),
            const _TransactionItem(name: "Vignesh K", plan: "Yearly Pack", amount: "₹12,000", date: "Mar 15, 10:00 AM", isSuccess: true),
            const SizedBox(height: 100),
          ],
        ),
      ),
    );
  }
}

class _SummaryCard extends StatelessWidget {
  final String title;
  final String amount;
  final IconData icon;
  final Color color;

  const _SummaryCard({required this.title, required this.amount, required this.icon, required this.color});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Card(
      elevation: 0,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24), side: BorderSide(color: theme.colorScheme.onSurface.withValues(alpha: 0.1))),
      color: theme.colorScheme.surface,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Icon(icon, color: color, size: 24),
            const SizedBox(height: 12),
            Text(title, style: theme.textTheme.labelSmall?.copyWith(color: theme.colorScheme.onSurface.withValues(alpha: 0.6))),
            const SizedBox(height: 4),
            Text(amount, style: theme.textTheme.titleMedium?.copyWith(fontWeight: FontWeight.w900)),
          ],
        ),
      ),
    );
  }
}

class _TransactionItem extends StatelessWidget {
  final String name;
  final String plan;
  final String amount;
  final String date;
  final bool isSuccess;

  const _TransactionItem({required this.name, required this.plan, required this.amount, required this.date, required this.isSuccess});

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Card(
      elevation: 0,
      margin: const EdgeInsets.symmetric(vertical: 4),
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      color: theme.colorScheme.surface,
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Row(
          children: [
            Container(
              width: 40,
              height: 40,
              decoration: BoxDecoration(
                color: (isSuccess ? const Color(0xFF10B981) : Colors.red).withValues(alpha: 0.1),
                shape: BoxShape.circle,
              ),
              child: Icon(
                isSuccess ? Icons.arrow_upward : Icons.history,
                color: isSuccess ? const Color(0xFF10B981) : Colors.red,
                size: 20,
              ),
            ),
            const SizedBox(width: 12),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(name, style: theme.textTheme.titleSmall?.copyWith(fontWeight: FontWeight.bold)),
                  Text(plan, style: theme.textTheme.bodySmall?.copyWith(color: theme.colorScheme.onSurface.withValues(alpha: 0.6))),
                ],
              ),
            ),
            Column(
              crossAxisAlignment: CrossAxisAlignment.end,
              children: [
                Text(amount, style: theme.textTheme.titleSmall?.copyWith(fontWeight: FontWeight.w900, color: isSuccess ? const Color(0xFF10B981) : Colors.red)),
                Text(date, style: theme.textTheme.labelSmall?.copyWith(color: theme.colorScheme.onSurface.withValues(alpha: 0.5))),
              ],
            ),
          ],
        ),
      ),
    );
  }
}
